build.gradle

```diff
@@ -397,21 +400,61 @@ task mapIntermediaryJar(dependsOn: [downloadMcLibs, build]) {
 	}
 }
 
-task mapNamedJar(dependsOn: mapIntermediaryJar) {
+task mapYarnJar(dependsOn: mapIntermediaryJar) {
+	group = mapJarGroup
 	inputs.files downloadMcLibs.outputs.files.files
-	outputs.file(namedJar)
+	outputs.file(yarnJar)
 
 	//Force the task to always run
 	outputs.upToDateWhen {false}
 
 	doLast {
-		logger.lifecycle(":mapping minecraft to named")
+		logger.lifecycle(":mapping minecraft to yarn")
 
 		def tinyInput = new File("build/libs/yarn-tiny-${yarnVersion}.gz")
-		mapJar(namedJar, intermediaryJar, tinyInput, libraries, "intermediary", "named")
+		mapJar(yarnJar, intermediaryJar, tinyInput, libraries, "intermediary", "named")
 	}
 }
 
+task buildTinyWithEnum(dependsOn: "mergeTiny", type: FileOutput) {
+	group = buildMappingGroup
+	def noEnum = mergeTiny.fileOutput
+	def namedWithEnum = new File(tempDir, "named-with-enum.tiny")
+	fileOutput = namedWithEnum
+
+	outputs.upToDateWhen { false }
+
+	doLast {
+		logger.lifecycle(":seeking auto-mappable fields")
+		String[] argsPropose = [
+			mergedFile.getAbsolutePath(), // must use official jar
+			noEnum.getAbsolutePath(),
+			namedWithEnum.getAbsolutePath(),
+			"--writeAll"
+		]
+
+		new CommandProposeFieldNames().run(argsPropose)
+	}
+}
+
+task mapNamedJar(dependsOn: [buildTinyWithEnum, mapIntermediaryJar]) {
+	group = mapJarGroup
+	inputs.files downloadMcLibs.outputs.files.files
+	outputs.file(namedJar)
+
+	//Force the task to always run
+	outputs.upToDateWhen { false }
+
+	doLast {
+		logger.lifecycle(":mapping minecraft to named")
+
+		def namedWithEnum = buildTinyWithEnum.fileOutput
```

Comment:
Dont really need this as its own def, not a big thing.

Reply:
Fixed.